### PR TITLE
Allow customizing logged software versions

### PR DESCRIPTION
**Pull request by Gallaecio**
Resolves: #4203

### CODE REVIEW

**Comment by wRAR** [comment]
Related: #4203

File: scrapy/utils/versions.py [hunk 1]
```diff
@@ -1,31 +1,48 @@
...
-        ("parsel", parsel.__version__),
-        ("w3lib", w3lib.__version__),
-        ("Twisted", twisted.version.short()),
-        ("Python", sys.version.replace("\n", "- ")),
-        ("pyOpenSSL", get_openssl_version()),
-        ("cryptography", cryptography.__version__),
-        ("Platform", platform.platform()),
-    ]
+def scrapy_components_versions(
+    components: list | None = None,
```
Comment (kmike) [inline scrapy/utils/versions.py:38]: What do you think about not adding this argument to scrapy_components_version? In this case, implementation will be different: it'd call `get_versions(_DEFAULT_SOFTWARE)`. We still should deprecate it probably.

File: scrapy/utils/log.py [hunk 2]
```diff
@@ -174,12 +175,11 @@ def log_scrapy_info(settings: Settings) -> None:
         "Scrapy %(version)s started (bot: %(bot)s)",
         {"version": scrapy.__version__, "bot": settings["BOT_NAME"]},
     )
-    versions = [
-        f"{name} {version}"
-        for name, version in scrapy_components_versions()
-        if name != "Scrapy"
-    ]
-    logger.info("Versions: %(versions)s", {"versions": ", ".join(versions)})
+    software = settings.getlist("LOG_VERSIONS")
```
Comment (Laerte) [inline scrapy/utils/log.py:178]: Looks good! Just wondering if this variable should be in plural.

Comment (Gallaecio) [reply scrapy/utils/log.py:178]: ”software” is plural, singular would be “pieces of software”, “software components”, or similar. I went with “items” to keep things short.

**Review by kmike — APPROVED** [review]

File: scrapy/utils/log.py (hunk 2 above)
Comment (Laerte) [reply scrapy/utils/log.py:178]: TIL, thank you @Gallaecio

**Review by Laerte — APPROVED** [review]

File: scrapy/utils/versions.py [hunk 3]
```diff
@@ -1,31 +1,46 @@
...
-        ("Twisted", twisted.version.short()),
-        ("Python", sys.version.replace("\n", "- ")),
-        ("pyOpenSSL", get_openssl_version()),
-        ("cryptography", cryptography.__version__),
-        ("Platform", platform.platform()),
-    ]
+    warn(
+        (
+            "scrapy.utils.versions.scrapy_components_versions is deprecated, "
+            "use scrapy.utils.versions.get_versions instead."
```
Comment (wRAR) [inline scrapy/utils/versions.py:41]:
```suggestion
            "scrapy.utils.versions.scrapy_components_versions() is deprecated, "
            "use scrapy.utils.versions.get_versions() instead."
```

File: scrapy/utils/versions.py [hunk 4]
```diff
@@ -1,31 +1,46 @@
...
+        ),
+        ScrapyDeprecationWarning,
+        stacklevel=2,
+    )
+    return get_versions(_DEFAULT_SOFTWARE)
```
Comment (wRAR) [inline scrapy/utils/versions.py:46]: Should this call just `get_versions()`, or should the `get_versions()` arg become mandatory? What's your plan for after `scrapy_components_versions()` is removed?

**Pull request merged**